src/dripper/polkadot/PolkadotActions.ts

```diff
@@ -1,223 +1,165 @@
-    logger.info("💸 teleporting tokens");
-
-    const dest = {
-      V3: {
-        interior: {
-          X1: {
-            Parachain: parachain_id,
-          },
-        },
-        parents: 0,
-      },
-    };
-
-    const addressHex = polkadotApi.registry.createType("AccountId", address).toHex();
-    const beneficiary = {
-      V3: {
-        interior: {
-          X1: {
-            AccountId32: { id: addressHex, network: null },
-          },
-        },
-        parents: 0,
-      },
-    };
-
-    const assets = {
-      V3: [
-        {
-          fun: { Fungible: dripAmount },
-          id: {
-            Concrete: {
-              interior: "Here",
-              parents: 0,
-            },
-          },
-        },
-      ],
-    };
-
-    const weightLimit = { Unlimited: null };
-
-    const feeAssetItem = 0;
-
-    const transfer = polkadotApi.tx.xcmPallet.limitedTeleportAssets(
-      dest,
-      beneficiary,
-      assets,
-      feeAssetItem,
-      weightLimit,
-    );
-
-    if (!this.account) throw new Error("account not ready");
-    const hash = await transfer.signAndSend(this.account, { nonce: -1 });
-
-    const result: DripResponse = { hash: hash.toHex() };
-    return result;
+  async teleportTokens(dripAmount: bigint, address: string, parachain_id: number): Promise<DripResponse> {
+    logger.info(`💸 teleporting tokens to ${address}, parachain ${parachain_id}`);
+
+    const addressBinary = Binary.fromBytes(encodeAccount(address));
+
+    const tx = await networkData.api.getTeleportTx({
+      dripAmount,
+      address: addressBinary,
+      parachain_id,
+      client,
+      nonce: this.#nonce++,
+    });
+
+    logger.debug(`Teleporting to ${address}: ${parachain_id}. Transaction ${tx}`);
+
+    // client.submit(tx) waits for the finalized value, which might be important for real money,
+    // but for the faucet drips, early respond is better UX
+    const submit$ = client.submitAndWatch(tx);
+
+    const hash = (await firstValueFrom(submit$.pipe(filter((value) => value.type === "broadcasted")))).txHash;
+
+    await firstValueFrom(submit$.pipe(filter((value) => value.type === "txBestBlocksState" && value.found)));
+
+    logger.info(`💸 teleporting tokens to ${address}, parachain ${parachain_id}: done: ${hash}`);
+    return { hash };
   }
 
-  async sendTokens(address: string, parachain_id: string, amount: bigint): Promise<DripResponse> {
+  async transferTokens(dripAmount: bigint, address: string): Promise<DripResponse> {
+    logger.info(`💸 sending tokens to ${address}`);
+    const tx = await networkData.api.getTransferTokensTx({
+      dripAmount,
+      address,
+      client,
+      nonce: this.#nonce++,
+    });
+    logger.debug(`Dripping to ${address}. Transaction ${tx}`);
+    // client.submit(tx) waits for the finalized value, which might be important for real money,
+    // but for the faucet drips, early respond is better UX
+
+    const submit$ = client.submitAndWatch(tx);
```

Comment:
```suggestion
    logger.debug(`Dripping to ${address}. Transaction ${tx}`);
    
    // client.submit(tx) waits for the finalized value, which might be important for real money,
    // but for the faucet drips, early respond is better UX
    const submit$ = client.submitAndWatch(tx);
```

Just a aesthetic change.